src/platform/w3dengine/client/w3dbibbuffer.cpp

```diff
@@ -0,0 +1,338 @@
+                        m_firstIndex = m_numBibIndices;
+                        m_firstVertex = m_numBibVertices;
+                    }
+
+                    for (int j = 0; j < m_numBibs; ++j) {
+                        if (!m_bibs[j].m_isUnused && m_bibs[j].m_hightlighted == (i != 0)) {
+
+                            int index = m_numBibVertices;
+                            int num_points = 4;
+
+                            if (m_numBibVertices + 6 >= m_maxBibVertex) {
+                                break;
+                            }
+
+                            int num_ind = 6;
+
+                            if (m_numBibIndices + 12 >= m_maxBibIndex) {
+                                break;
+                            }
+
+                            for (int k = 0; k < num_points; ++k) {
+                                float v;
+                                float u;
+
+                                Vector3 point(m_bibs[j].m_points[k]);
+                                switch (k) {
+                                    case 0:
+                                        u = 0.0;
+                                        v = 1.0;
+                                        break;
+                                    case 1:
+                                        u = 1.0;
+                                        v = 1.0;
+                                        break;
+                                    case 2:
+                                        u = 1.0;
+                                        v = 0.0;
+                                        break;
+                                    case 3:
+                                        u = 0.0;
+                                        v = 0.0;
+                                        break;
+                                    default:
+                                        break;
+                                }
+
+                                vertices->u1 = u;
+                                vertices->v1 = v;
+                                vertices->x = point.X;
+                                vertices->y = point.Y;
+                                vertices->z = point.Z;
+                                vertices->diffuse = color;
+
+                                ++vertices;
+                                ++m_numBibVertices;
+                            }
+
+                            *indices = index + 0;
+                            ++indices;
+                            *indices = index + 1;
+                            ++indices;
+                            *indices = index + 2;
+                            ++indices;
+                            *indices = index + 0;
+                            ++indices;
+                            *indices = index + 2;
+                            ++indices;
+                            *indices = index + 3;
+                            ++indices;
+
+                            m_numBibIndices += num_ind;
```

Comment:
I am confused by adding 6 here, when further above it tests with `m_numBibIndices + 12`

Reply:
dunno why its like that, can confirm its correct to binary